app/code/Magento/PricingStorefront/Model/PriceManagement.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\PricingStorefront\Model;
+
+use Psr\Log\LoggerInterface;
+
+class PriceManagement
+{
+    /**
+     * @var PriceRepository
+     */
+    private $priceRepository;
+
+    /**
+     * @var PriceBookRepository
+     */
+    private $priceBookRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param PriceRepository $priceRepository
+     * @param PriceBookRepository $priceBookRepository
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        PriceRepository $priceRepository,
+        PriceBookRepository $priceBookRepository,
+        LoggerInterface $logger
+    ) {
+        $this->priceRepository = $priceRepository;
+        $this->priceBookRepository = $priceBookRepository;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Remove prices.
+     *
+     * @param string $priceBookId
+     * @param array $productIds
+     * @return int
+     * @throws \Throwable
+     */
+    public function unassignPrices(string $priceBookId, array $productIds): int
+    {
+        return $this->priceRepository->delete($priceBookId, $productIds);
+    }
+
+    /**
+     * Assign product price to price book.
+     *
+     * @param string $bookId
+     * @param array $price
+     * @param string|null $parentId
+     * @return array
+     */
+    public function assignPrice(string $bookId, array $price, ?string $parentId = null) : array
+    {
+        $priceSearch = [];
+        $priceUpdate = [];
+        $finalKey = PriceRepository::KEY_FINAL;
+        $regularKey = PriceRepository::KEY_REGULAR;
+        // @TODO if qty not present what default value should be???
+        $qty = $price[PriceRepository::KEY_QTY] ?? 1.0000;
+        $productId = $price[PriceRepository::KEY_PRODUCT_ID] ?? null;
+
+        foreach ($price as $key => $value) {
+            if (in_array($key, $this->getPriceType())) {
+                $priceUpdate[$key . '_' . $finalKey] = $value[$finalKey] ?? null;
+                $priceUpdate[$key . '_' . $regularKey] = $value[$regularKey] ?? null;
+            }
+        }
+
+        if (empty($productId) || empty(array_filter($priceUpdate))) {
+            return $priceSearch;
+        }
+
+        $origBookId = $bookId;
+        $found = false;
+
+        if ($parentId) {
+            while (!$found && $bookId && $origBookId) {
+                try {
+                    $priceSearch = $this->getPriceRow($bookId, $productId, $qty, $parentId);
+                } catch (\Throwable $e) {
+                    $priceSearch = [];
+                }
+
+                if (!empty($priceSearch['price']) && $origBookId === $bookId) {
+                    // if new price different as actual but same as parent price - remove actual
```

Comment:
The same potential problem, what we have in "compare price with existing parent price"
Maybe it's better to remove this and simplify price assignment ?